src/runner.rs

```diff
@@ -171,26 +172,47 @@ impl<E: EnvController> Runner<E> {
 
     async fn run_single_case(&self, db: &E::DB, path: &PathBuf) -> Result<bool> {
         let case_path = path.with_extension(&self.config.test_case_extension);
-        let case = TestCase::from_file(case_path, &self.config).await?;
-        let output_path = path.with_extension(&self.config.output_result_extension);
-        let mut output_file = Self::open_output_file(&output_path).await?;
+        let case = TestCase::from_file(&case_path, &self.config).await?;
+        let result_path = path.with_extension(&self.config.result_extension);
+        let mut result_file = OpenOptions::new()
+            .create(true)
+            .write(true)
+            .read(true)
+            .open(&result_path)
+            .await?;
+
+        // Read old result out for compare later
+        let mut old_result = String::new();
+        result_file.read_to_string(&mut old_result).await?;
 
+        // Execute testcase
+        let mut new_result = Cursor::new(Vec::new());
         let timer = Instant::now();
-        case.execute(db, &mut output_file).await?;
+        case.execute(db, &mut new_result).await?;
         let elapsed = timer.elapsed();
 
-        output_file.flush().await?;
-        let is_different = self.compare(&path).await?;
-        if !is_different {
-            remove_file(output_path).await?;
+        // Truncate and write new result back
+        let mut result_file = OpenOptions::new()
```

Comment:
I try `result_file.set_len(0)`, but it isn't truncated so I open it again with truncate flag.

Reply:
You may need to `seek` to the start after `set_len()` to reset the file cursor.

Reply:
👍 That's the point.